kratos/geometries/quadrilateral_3d_9.h

```diff
@@ -388,40 +388,52 @@ template<class TPointType> class Quadrilateral3D9 : public Geometry<TPointType>
         return( std::sqrt( d[0]*d[0] + d[1]*d[1] + d[2]*d[2] ) );
     }
 
-    /**
-     * :TODO: the charactereistic sizes have to be reviewed
-     * by the one who is willing to use them!
-     */
-    /**
-     * This method calculates and returns area or surface area of
+    /** This method calculates and returns area or surface area of
      * this geometry depending to it's dimension. For one dimensional
      * geometry it returns zero, for two dimensional it gives area
      * and for three dimensional geometries it gives surface area.
+     *
      * @return double value contains area or surface
-     * area.
+     * area.N
      * @see Length()
      * @see Volume()
      * @see DomainSize()
+     * @todo could be replaced by something more suitable (comment by janosch)
      */
     double Area() const override
     {
-        Vector d = this->Points()[2] - this->Points()[0];
-        return( std::sqrt( d[0]*d[0] + d[1]*d[1] + d[2]*d[2] ) );
-    }
+        Vector temp;
+        this->DeterminantOfJacobian( temp, msGeometryData.DefaultIntegrationMethod() );
+        const IntegrationPointsArrayType& integration_points = this->IntegrationPoints( msGeometryData.DefaultIntegrationMethod() );
+        double Area = 0.0;
 
+        for ( unsigned int i = 0; i < integration_points.size(); i++ ) {
+            Area += temp[i] * integration_points[i].Weight();
+        }
 
-    double Volume() const override
-    {
-        Vector d = this->Points()[2] - this->Points()[0];
-        return( std::sqrt( d[0]*d[0] + d[1]*d[1] + d[2]*d[2] ) );
+        return Area;
     }
 
     /**
-     * :TODO: the charactereistic sizes have to be reviewed
-     * by the one who is willing to use them!
+     * This method calculates and returns the volume of this geometry.
+     * This method calculates and returns the volume of this geometry.
+     *
+     * This method uses the V = (A x B) * C / 6 formula.
+     *
+     * @return double value contains length, area or volume.
+     *
+     * @see Length()
+     * @see Area()
+     * @see Volume()
+     *
+     * @todo might be necessary to reimplement
```

Comment:
Those comments are not really reassuring this works...
Are there tests?

Reply:
These comments are old comments in Doxygen format